### ISSUE TITLE
Check line-endings before import

### ISSUE THREAD

**Issue by GaryJones** [Automattic/WPCOM-Legacy-Redirector#52]
tl;dr Check line-endings in a CSV file are consistent with the host environment before import.

----

I tried to use this plugin with a `.csv` file that produced errors on import. What was reported had overlapping URLs, and only showed 7 out of the 88 URLs attempting to be imported.

I was able to spot a pattern on which URLs these were in the CSV file, and found it was about every 2000 bytes. That lead me to consider that the `fgetcsv()` file appeared to be [grabbing 2000 bytes](https://github.com/Automattic/WPCOM-Legacy-Redirector/blob/1.3.0/includes/wp-cli.php#L98) of data as a single row, which implied that multiple rows were being concatenated into one.

From this, I was able to make and prove the conclusion that the file was using Mac OS 9 line-endings (`CR`). The server I was trying to run this on required Unix line-endings (`LF`). Converting the file locally, pushing to the server and importing then worked fine.

The plugin should check for line-endings, or, as suggested [here](https://www.php.net/manual/en/function.fgetcsv.php#87196), enable auto-detection of line-endings before opening the file.

**Comment by bdtech** [comment]
@GaryJones Good call spotting this - let's make it better.

Do you still by any chance have the problematic CSV handy? It would be an excellent test case. In the mean time I'll create a few samples file's with various line-endings.

**Comment by GaryJones** [comment]
I've shared a file with @bdtech to use as an example.

**Comment by GaryJones** [comment]
Closed via #55.
